llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -2641,6 +2641,31 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
       return BinaryOperator::CreateFMulFMF(Src, Select, II);
     }
 
+    // ldexp(x, c ? exp : 0) -> c ? ldexp(x, exp) : x
+    // ldexp(x, c ? 0 : exp) -> c ? x : ldexp(x, exp)
+    ///
```

Comment:
Does this pattern exist in real-world cases?

Reply:
I found it in math library code for scale if out of range

Reply:
Anyway, I have no objections since it is the right canonicalization as we already did for other fp intrinsics.